drivers/ethernet/Kconfig

```diff
@@ -11,6 +11,9 @@ menuconfig ETH_DRIVER
 
 if ETH_DRIVER
 
+config RNG
+	default y
```

Comment:
You cannot do this, this will completely bypass dependencies. As an example, take a board that does not have support for rng, e.g. `niosv_g`, configure hello world for it, run menuconfig, go to RNG and confirm that the option cannot be selected:
```
  config RNG
        bool "Enable inclusion of RNG functions"
        default y
        imply ENTROPY_GENERATOR(=n)
        depends on ENTROPY_HAS_DRIVER(=n) || TEST_RANDOM_GENERATOR(=n)
        help
          Subsystems and applications that need to use the sys_rand_get() set
          of APIs will set this config.

          ENTROPY_HAS_DRIVER is true if the DT is enabled.
```
Now go and enable networking and ethernet, this now selects RNG even though the option's requirements have not been met

Reply:
Then it wont turn on. If you don't have entropy and you haven't turned on TEST_RANDOM_GENERATOR then you can't use RNG.

Reply:
> Then it wont turn on

It does though, because select **bypasses** Kconfig dependencies

Reply:
I see the same thing but it kind of operates as expected. The actual selection of the RNG/CSRNG doesn't show up until you select the "Allow non-random number generator". When you do then you can select RNG/CSRNG. If you unselect the non-random selection, the RNG/CSRNG goes away. 

I guess the question here though is if that makes sense. Should the "non-random number generator" (which is really non-random number entropy) be under entropy?